provider/plan.go

```diff
@@ -131,15 +131,28 @@ func (s *RawProviderServer) PlanResourceChange(ctx context.Context, req *tfproto
 	if err != nil {
 		return resp, fmt.Errorf("failed to determine resource type ID: %s", err)
 	}
+
 	if !objectType.Is(tftypes.Object{}) {
-		// this is not a valid resource type - likely a freeform CR without schema
+		// non-structural resources have no schema so we just use the
+		// type information we can get from the config
+		objectType = ppMan.Type()
+
+		// TODO dry run here to catch any types added on the server side
```

Comment:
Did you run into any technical troubles trying to do the dry-run?

IMO, bar any weird issues, we should really do the dry-run here, not necessarily for catching default values (there can hardly be any for these CRDs since there is no schema), but more crucially for catching syntax issues in time. Especially because we have no type reference here, we're very exposed to invalid user input blowing up later in the apply. The dry-run call will catch these and allow us to instead print a controlled error explaining the issue during plan rather than fail of the apply (and leave tainted state).